portfolio/src/main/webapp/request.js

```diff
@@ -25,4 +25,109 @@ class ScheduleRequest {
   }
 }
 
-// TODO(tomasalvarez): write function to fetch results and display results
+// Once a call is made to the server for scheduling this array will be a local
+// copy of the results.
+var scheduledTasks = []
+
+    /**
+     * Gets all of the scheduling information from the UI and returns a
+     * ScheduleRequest with all of the data.
+     */
+    function
+    createRequestFromUiInformation() {
+      const startTime = document.getElementById('working-hour-start').value;
+      const endTime = document.getElementById('working-hour-end').value;
+      const events = collectAllEvents();
+      const tasks = collectAllTasks();
+      const algorithmType = document.getElementById('algorithm-type').value;
+      const scheduleRequest = new ScheduleRequest(
+          events, tasks, getTimeObject(startTime), getTimeObject(endTime),
+          algorithmType);
+      return scheduleRequest;
+    }
+
+/**
+ * Gets called when the user hits 'Start Scheduling'.
+ */
+function
+startScheduling() {
+  const scheduleRequest = createRequestFromUiInformation();
+  // Create the request to send to the server using the data we collected from
+  // the web form.
+  fetchScheduledTasksFromServlet(scheduleRequest).then((scheduledTaskArray) => {
+    updateResultsOnPage(scheduledTaskArray);
+  });
+}
+
+/**
+ * Updates the UI to show the results of a query.
+ */
+function updateResultsOnPage(scheduledTaskArray) {
+  const resultElement = document.getElementById('scheduled-task-list');
+  resultElement.innerHTML = '';
+  scheduledTaskArray.forEach(addScheduledTaskToPage, resultElement);
+}
+
+/**
+ * Handles sending the ScheduleRequest object to the servlet and returns the
+ * scheduled tasks.
+ */
+function
+fetchScheduledTasksFromServlet() {
+  const scheduleRequest = createRequestFromUiInformation();
+  const json = JSON.stringify(scheduleRequest);
+  return fetch('/schedule', {method: 'POST', body: json})
```

Comment:
I wonder if you need this return on this line? Would the final `return response.json()` return the Json object there?

Reply:
I kind of forgot if this needs to be a return or not but the walkthroughs put a return here so I did too.

Reply:
Ya you still need to `return` here, because the result of this function is that you're returning a promise to the caller. You can think of a promise just like any other object that needs to be returned. It's exactly the same as if say you wanted to transform and return an array of something like:
```
function foo() {
  var array = [1, 2, 3];
  return array.map((i) => { return i + 1; });
}
```
returns `[2,3,4]`.